src/Drupal/Commands/core/FieldCreateCommands.php

```diff
@@ -0,0 +1,679 @@
+use Drupal\content_translation\ContentTranslationManagerInterface;
+use Drupal\Core\Entity\Display\EntityDisplayInterface;
+use Drupal\Core\Entity\Display\EntityFormDisplayInterface;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityReferenceSelection\SelectionPluginManagerInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Field\FieldTypePluginManagerInterface;
+use Drupal\Core\Field\WidgetPluginManager;
+use Drupal\Core\Url;
+use Drupal\field\FieldConfigInterface;
+use Drupal\field\FieldStorageConfigInterface;
+use Drush\Commands\DrushCommands;
+use Symfony\Component\Console\Input\InputOption;
+
+class FieldCreateCommands extends DrushCommands implements CustomEventAwareInterface
+{
+    use CustomEventAwareTrait;
+
+    /** @var FieldTypePluginManagerInterface */
+    protected $fieldTypePluginManager;
+    /** @var WidgetPluginManager */
+    protected $widgetPluginManager;
+    /** @var SelectionPluginManagerInterface */
+    protected $selectionPluginManager;
+    /** @var EntityTypeManagerInterface */
+    protected $entityTypeManager;
+    /** @var EntityTypeBundleInfoInterface */
+    protected $entityTypeBundleInfo;
+    /** @var EntityFieldManagerInterface */
+    protected $entityFieldManager;
+    /** @var ModuleHandlerInterface */
+    protected $moduleHandler;
+    /** @var ContentTranslationManagerInterface */
+    protected $contentTranslationManager;
+
+    public function __construct(
+        FieldTypePluginManagerInterface $fieldTypePluginManager,
+        WidgetPluginManager $widgetPluginManager,
+        SelectionPluginManagerInterface $selectionPluginManager,
+        EntityTypeManagerInterface $entityTypeManager,
+        EntityTypeBundleInfoInterface $entityTypeBundleInfo,
+        ModuleHandlerInterface $moduleHandler,
+        EntityFieldManagerInterface $entityFieldManager
+    ) {
+        $this->fieldTypePluginManager = $fieldTypePluginManager;
+        $this->widgetPluginManager = $widgetPluginManager;
+        $this->selectionPluginManager = $selectionPluginManager;
+        $this->entityTypeManager = $entityTypeManager;
+        $this->entityTypeBundleInfo = $entityTypeBundleInfo;
+        $this->moduleHandler = $moduleHandler;
+        $this->entityFieldManager = $entityFieldManager;
+    }
+
+    public function setContentTranslationManager($manager): void
+    {
+        $this->contentTranslationManager = $manager;
+    }
+
+    /**
+     * Create a new field
+     *
+     * @command field:create
+     * @aliases field-create,fc
+     *
+     * @validate-entity-type-argument entityType
+     * @validate-optional-bundle-argument entityType bundle
```

Comment:
@DieterHolvoet ^

Reply:
That's still a leftover from where I took the command, it's an abstraction for validating entity type/bundle arguments: https://github.com/wieni/wmscaffold/blob/release/v1/src/Commands/ValidatorsCommands.php

It's used in other commands I mentioned before: `field:delete`, `field:info`, `base-field:info`, `base-field-override:create`. If we're planning on incorporating those commands as well, it might be worth it to keep it abstract, if not we can copy/paste it into the class.

Reply:
There are no changes to ValidatorsCommands in the PR so those annotations don't yet do anything. I'm OK with the abstraction, I just think they should have their corresponding code in Drush.